server/src/main/java/org/jboss/as/server/BootstrapListener.java

```diff
@@ -104,10 +105,26 @@ private void done(final long bootstrapTime, final StabilityStatistics statistics
         final int problem = statistics.getProblemsCount();
         final int started = statistics.getStartedCount();
         if (failed == 0 && problem == 0) {
-            ServerLogger.AS_ROOT_LOGGER.startedClean(prettyVersion, bootstrapTime, started, active + passive + onDemand + never + lazy, onDemand + passive + lazy);
+            String append = "";
+            if (configFiles != null & !configFiles.isEmpty() & isHostController & isMaster & configFiles.size() == 2) {
+                append = ServerLogger.AS_ROOT_LOGGER.hostControllerMasterConfigFilesInUse(configFiles.get(0), configFiles.get(1));
+            } else if (configFiles != null & !configFiles.isEmpty() & isHostController & !isMaster & configFiles.size() == 1) {
+                append = ServerLogger.AS_ROOT_LOGGER.hostControllerSlaveConfigFileInUse(configFiles.get(0));
+            } else if (configFiles != null & !configFiles.isEmpty() & !isHostController & configFiles.size() == 1) {
+                append = ServerLogger.AS_ROOT_LOGGER.serverConfigFileInUse(configFiles.get(0));
+            }
```

Comment:
This piece of code is repeated in an `if/else`. You can extract it from that flow control structure to avoid redundancy.
In addition, && is preferred here since it will break the shortcut as soon as it detects on operand as false, instead of & that will evaluate all the operands aways.

Reply:
Removing the `isMaster` argument, you could arrive at something like the following:

```
String append = "";
if (configFiles != null && !configFiles.isEmpty()) {
    if (isHostController) {
        if (configFiles.size() == 2)
          //  master host contoller message
        else {
          //  slave host controller message
        }
    } else {
        // server message
    }
}
```

That will facilitate the debugging and it is more redeable